zulip/integrations/bridge_with_matrix/matrix_bridge.py

```diff
@@ -76,12 +82,12 @@ def _matrix_to_zulip(room: Any, event: Dict[str, Any]) -> None:
                     "subject": zulip_config["topic"],
                     "content": content,
                 })
-            except MatrixRequestError as e:
+            except Exception as e:  # XXX This should be more specific
                 # Generally raised when user is forbidden
-                raise Exception(e)
+                raise Bridge_ZulipException(e)
             if result['result'] != 'success':
                 # Generally raised when API key is invalid
-                raise Exception(result['msg'])
+                raise Bridge_ZulipException(result['msg'])
```

Comment:
We should check the `result['code']` to see whether it's actually an invalid API key, here.  

(If it there is no `code`, I'll add one to the server).

Reply:
I'm not sure where to look for this - @rheaparekh can you comment on this, since you added this exception section in `93627550`?

Reply:
There is no `code` here. We get the result as `{'msg': 'Invalid API key', 'result': 'error'}`. I just tested and confirmed this again.

Reply:
OK.  I'll integrate adding one; basically, other errors that might occur here feel more transient.